From 37f530ec4e29fce90a42af7558ad4ee638bb3b40 Mon Sep 17 00:00:00 2001
From: sam_chen <sam_chen@asus.com>
Date: Tue, 2 Apr 2013 10:37:09 +0800
Subject: [PATCH] Audio: soc: wcd9310: add check bandgap status function.

Export function to headset driver to know whether we should disable
badgap power or not after headset detection.

Change-Id: I1b890c559a26537f510dfabed3cfec16120f59de
Signed-off-by: sam_chen <sam_chen@asus.com>
---
 sound/soc/codecs/wcd9310.c | 7 +++++++
 sound/soc/codecs/wcd9310.h | 1 +
 2 files changed, 8 insertions(+)

diff --git a/sound/soc/codecs/wcd9310.c b/sound/soc/codecs/wcd9310.c
index 1c5c615..e5012d7 100644
--- a/sound/soc/codecs/wcd9310.c
+++ b/sound/soc/codecs/wcd9310.c
@@ -4066,6 +4066,13 @@ static void tabla_shutdown(struct snd_pcm_substream *substream,
 	}
 }
 
+int tabla_check_bandgap_status(struct snd_soc_codec *codec)
+{
+	struct tabla_priv *tabla = snd_soc_codec_get_drvdata(codec);
+
+	return tabla->bandgap_type;
+}
+
 int tabla_mclk_enable(struct snd_soc_codec *codec, int mclk_enable, bool dapm)
 {
 	struct tabla_priv *tabla = snd_soc_codec_get_drvdata(codec);
diff --git a/sound/soc/codecs/wcd9310.h b/sound/soc/codecs/wcd9310.h
index b002316a..c10529c 100644
--- a/sound/soc/codecs/wcd9310.h
+++ b/sound/soc/codecs/wcd9310.h
@@ -180,6 +180,7 @@ struct tabla_mbhc_config {
 	/* swap_gnd_mic returns true if extern GND/MIC swap switch toggled */
 	bool (*swap_gnd_mic) (struct snd_soc_codec *);
 };
+extern int tabla_check_bandgap_status(struct snd_soc_codec *codec);
 
 extern int tabla_hs_detect(struct snd_soc_codec *codec,
 			   const struct tabla_mbhc_config *cfg);
